Language/Expressions/Constants/as_type_A01_t02.dart

```diff
@@ -73,7 +73,8 @@ main() {
   const fo7 = IntET2(1) as FutureOr<IntET1Alias>;
 
   const f1 = foo as void Function();
-  const f2 = bar<int> as int Function<int>(int);
+  const f2 = as as void Function();
+  const f3 = bar<int> as int Function<int>(int);
```

Comment:
If this is not intended to be a compile-time error I think we'd do this:
```suggestion
  const f3 = bar<int> as int Function(int);
```

Reply:
Updated